### PR TITLE
Bug 1125708 - [settings] refactor homescreen panel with AMD pattern

### PR DESCRIPTION
**Pull request by EragonJ**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f8da7889f02e)

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=06c9549a1c33)

File: apps/settings/elements/homescreen.html [hunk 1]
```diff
@@ -23,7 +23,8 @@ <h2 data-l10n-id="vertical-homescreen-layout">Layout</h2>
         </ul>
```
Comment (crh0716) [inline apps/settings/elements/homescreen.html]: nit: Let's also remove the texts from the html file.

Comment (EragonJ) [inline apps/settings/elements/homescreen.html]: kk !

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7c6cc8831808)

File: apps/settings/js/panels/homescreen/panel.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          Homescreen.cols = this.value;
+        });
+
+        // When initializing, this value may be null
+        if (Homescreen.cols !== null) {
+          this._updateCols(Homescreen.cols);
+        }
+
+        Homescreen.observe('cols', (newValue, oldValue) => {
+          if (newValue !== oldValue) {
```
Comment (crh0716) [inline apps/settings/js/panels/homescreen/panel.js]: `Observable` does the check for you so this is not required.

File: apps/settings/js/panels/homescreen/panel.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        this._elements = {};
+        this._elements.gridSelect =
+          panel.querySelector('[name="grid.layout.cols"]');
+
+        this._elements.gridSelect.addEventListener('change', function() {
+          Homescreen.cols = this.value;
+        });
+
+        // When initializing, this value may be null
+        if (Homescreen.cols !== null) {
```
Comment (crh0716) [inline apps/settings/js/panels/homescreen/panel.js]: It seems you already do the check in `_updateCols`.

File: apps/settings/js/panels/homescreen/homescreen.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+define(function(require) {
+  'use strict';
+
+  var VerticalPreferences = require('shared/homescreens/vertical_preferences');
+  var Observable = require('modules/mvvm/observable');
+
+  var Homescreen = Observable({
```
Comment (crh0716) [inline apps/settings/js/panels/homescreen/homescreen.js]: Please provide a function for setting the value of cols as the action is asynchronous.

File: apps/settings/js/panels/homescreen/panel.js (hunk 2 above)
Comment (EragonJ) [inline apps/settings/js/panels/homescreen/panel.js]: Soga !! Thanks

File: apps/settings/js/panels/homescreen/panel.js (hunk 3 above)
Comment (EragonJ) [inline apps/settings/js/panels/homescreen/panel.js]: okay !!!

File: apps/settings/js/panels/homescreen/homescreen.js (hunk 4 above)
Comment (EragonJ) [inline apps/settings/js/panels/homescreen/homescreen.js]: ok !!

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=95b48f4d7bec)

**Pull request merged**